### PR TITLE
Add declaration for scopes of transaction related variables and configurations (#5725)

### PR DESCRIPTION
**Pull request by ti-srebot**
cherry-pick #5725 to release-3.0
You can switch your code base to this Pull Request by using [git-extras](https://github.com/tj/git-extras):
```bash
# In docs-cn repo:
git pr https://github.com/pingcap/docs-cn/pull/5732
```

After apply modifications, you can push your change to this PR via:
```bash
git push git@github.com:ti-srebot/docs-cn.git pr/5732:release-3.0-868bf60ddd72
```

---

### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

There are 3 variables and configurations to control the retrial of transactions:
- Optimistic transactions: `tidb_disable_txn_auto_retry` and `tidb_retry_limit`.
- Pessimistic transactions: `max-retry-count`.

However, the document doesn't clarify the scopes of each variable or configuration. I added the clarification of each variable and configuration to the document.

### Which TiDB version(s) do your changes apply to? (Required)

- [x]  master (the latest development version)
- [x] v5.0 (TiDB 5.0 versions)
- [x] v4.0 (TiDB 4.0 versions)
- [x] v3.1 (TiDB 3.1 versions)
- [x] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

N/A

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [x] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by TomShawn** [comment]
@djshow832 PTAL whether there is any version-specific change to make, thanks.

File: system-variables.md [hunk 1]
```diff
@@ -60,7 +231,108 @@ MySQL 系统变量 (System Variables) 是一些系统参数，用于调整数据
 
 * 会话范围的系统变量仅仅会在创建会话时才会根据全局范围系统变量初始化自己的值。更改全局范围的系统变量不会改变已经创建的会话正在使用的系统变量的值。
```
Comment (TomShawn) [inline system-variables.md:232]:
```suggestion
* 会话范围的系统变量仅仅会在创建会话时才会根据全局范围系统变量初始化自己的值。更改全局范围的系统变量不会改变已经创建的会话正在使用的系统变量的值。 
```

**Comment by djshow832** [comment]
@TomShawn  The conflicts are resolved.

**Pull request merged**